collector/lib/CollectorStatsExporter.cpp

```diff
@@ -168,6 +152,12 @@ void CollectorStatsExporter::run() {
   prometheus::Gauge* lineage_std_dev = &collectorProcessLineageInfo.Add({{"type", "std_dev"}});
   prometheus::Gauge* lineage_avg_string_len = &collectorProcessLineageInfo.Add({{"type", "lineage_avg_string_len"}});
 
+  // Following counters are used for detailed metrics
+  prometheus::Gauge* userspaceEvents;
+  prometheus::Family<prometheus::Gauge>* collectorTypedEventCounters;
+  prometheus::Family<prometheus::Gauge>* collectorTypedEventTimesTotal;
+  prometheus::Family<prometheus::Gauge>* collectorTypedEventTimesAvg;
```

Comment:
We might want to ensure these are initialized as `nullptr`, doesn't seem like it could reach a point where they are used uninitialized, but better be safe than sorry.
```suggestion
  prometheus::Gauge* userspaceEvents = nullptr;
  prometheus::Family<prometheus::Gauge>* collectorTypedEventCounters = nullptr;
  prometheus::Family<prometheus::Gauge>* collectorTypedEventTimesTotal = nullptr;
  prometheus::Family<prometheus::Gauge>* collectorTypedEventTimesAvg = nullptr;
```